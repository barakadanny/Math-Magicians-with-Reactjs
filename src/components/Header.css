header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #333;
}

nav ul {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

nav ul li {
  color: #fff;
  text-decoration: none;
  padding: 10px;
}
