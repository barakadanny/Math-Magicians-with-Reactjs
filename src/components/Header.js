import React from 'react';
import { Link } from 'react-router-dom';

function Header() {
  return (
    <header>
      <h1>Math Magician</h1>
      <nav>
        <ul>
          <li>
            <Link to='/'>Home</Link>
          </li>
          <li>
            <Link to='/calculator'>Calculator</Link>
          </li>
          <li>
            <Link to='/quote'>Quote</Link>
          </li>
        </ul>
      </nav>
    </header>
  );
}

export default Header;
