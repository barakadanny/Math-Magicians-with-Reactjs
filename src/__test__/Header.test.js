import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Calculator';
import '@testing-library/jest-dom';

describe('Test Header page', () => {
  it('testing Header page UI rendering', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
