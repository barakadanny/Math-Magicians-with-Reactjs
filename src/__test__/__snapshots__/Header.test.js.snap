// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header page testing Header page UI rendering 1`] = `
<div
  className="calculator-container"
>
  <div
    className="calculator-usage"
  >
    Lets do some Math!
  </div>
  <div
    className="calculator"
  >
    <div
      className="result"
    >
      <p>
        0
      </p>
    </div>
    <span
      onClick={[Function]}
      role="presentation"
    >
      AC
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      +/-
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      %
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      รท
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      7
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      8
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      9
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      x
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      4
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      5
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      6
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      -
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      1
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      2
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      3
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      +
    </span>
    <span
      className="double"
      onClick={[Function]}
      role="presentation"
    >
      0
    </span>
    <span
      onClick={[Function]}
      role="presentation"
    >
      .
    </span>
    <span
      className="operator"
      onClick={[Function]}
      role="presentation"
    >
      =
    </span>
  </div>
</div>
`;
